### PR TITLE
Remove PRE_60_NODE_CHECKPOINT

### PR DESCRIPTION
**Pull request by DaveCTurner**
This commit removes the obsolete `PRE_60_NODE_CHECKPOINT` constant for dealing
with 5.x nodes' lack of sequence number support.

Backported to 7.x as #42531.

### CODE REVIEW

**Comment by DaveCTurner** [comment]
The parallel change in 7.x is a little more involved so I will open a separate PR for that.

File: server/src/main/java/org/elasticsearch/index/seqno/ReplicationTracker.java [hunk 1]
```diff
@@ -911,10 +907,6 @@ public synchronized void markAllocationIdAsInSync(final String allocationId, fin
     }
 
     private boolean updateLocalCheckpoint(String allocationId, CheckpointState cps, long localCheckpoint) {
-        // a local checkpoint of PRE_60_NODE_CHECKPOINT cannot be overridden
-        assert cps.localCheckpoint != SequenceNumbers.PRE_60_NODE_CHECKPOINT ||
-            localCheckpoint == SequenceNumbers.PRE_60_NODE_CHECKPOINT :
-            "pre-6.0 shard copy " + allocationId + " unexpected to send valid local checkpoint " + localCheckpoint;
         // a local checkpoint for a shard copy should be a valid sequence number or the pre-6.0 sequence number indicator
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/index/seqno/ReplicationTracker.java]: update comment please and change assertion below to `assert localCheckpoint >= SequenceNumbers.NO_OPS_PERFORMED :`

**Review by ywelsch — APPROVED** [review]
One comment, looking good o.w.

**Review by dnhatn — APPROVED** [review]
LGTM

**Comment by DaveCTurner** [comment]
I opened #42532 as I don't think the build failure relates to this change.

@elasticmachine please run elasticsearch-ci/1

**Comment by DaveCTurner** [comment]
Not sure what's going on here, let's try one more time.

@elasticmachine please run elasticsearch-ci/1

**Comment by DaveCTurner** [comment]
@elasticmachine please run elasticsearch-ci/2

**Pull request merged**